src/appFunctions.js

```diff
@@ -12,23 +11,19 @@ export function getFunctionFindCard(key) {
 }
 
 export function findCardById(id, cards) {
-    
     return getFunctionFindCard('_id')(id, cards);
 }
 
 export function findCardByTitle(title, cards) {
-    
     return getFunctionFindCard('title')(title, cards);
 }
 
 export function findCardsByStatus(status, cards) {
-    
     return cards[status] || [];
 }
 
 export function findCardsByDisabledStatuses(statuses, cards) {
-    
-    const cardsWithStatusDisabledOrDeleted = statuses.reduce((acc, status) => {
+    return Object.values(statuses.reduce((acc, status) => {
```

Comment:
now it's not readable
try to make it easier and why do you need reduce here?

Reply:
The findCardsByDisabledStatuses function return cards which have disabled status

statuses = [
    {name: Open, enabled: true ....},
    {name: In Progress, enabled: false ....},
    {name: Closed, enabled: true ....}
]

cards = {
    Open: [...],
    In Progress: [...],
    Closed: [...]
}

I need to leave only those cards in which statuses are disabled
in the method i remove the fields that enabled

after work reduce method:

cards = {    
    In Progress: [...]    
}

after Object.values:

cards = { [ [...] ]}

after flat: 

cards = {[...]}

Reply:
how would you get cards whose statuses are disabled or nonexistent?
My experience is weak)

Reply:
It is also important to consider that there may be fields of deleted statuses. They must also be left and displayed in Other

Reply:
ok
don't wrap all expression with `Object.values` keep it in separate variable
you don't need to do `let cards = {...acc};` every time instead you can pass new object in reduce on initialization

Reply:
export function findCardsByDisabledStatuses(statuses, cards) {
    const cardsDisabledOrDeletedStatus = statuses.reduce((acc, status) => {
        let cards = {...acc};

        if (status.enabled) {
            delete cards[status.name];
        }

        return cards;
    }, cards);
    
    return Object.values(cardsDisabledOrDeletedStatus).flat();
}

I don't know how to simplify
I tried like this

export function findCardsByDisabledStatuses(statuses, cards) {
    const cardsDisabledOrDeletedStatus = statuses.reduce((acc, status) => {
       if (status.enabled) {
            delete acc[status.name];
        }

        return acc;
    }, cards);
    
    return Object.values(cardsDisabledOrDeletedStatus).flat();
}

But it doesn't work

Reply:
I need help

Reply:
export function findCardsByDisabledStatuses(statuses, cards) {
  const reducedCards = statuses.reduce((acc, status) => {
    if (status.enabled) {
      delete cards[status.name];
    }

    return cards;
  }, { ...cards });

  return Object.values(reducedCards).flat()
}

try it

Reply:
```
export function findCardsByDisabledStatuses(statuses, cards) {
  const reducedCards = statuses.reduce((acc, status) => {
    if (status.enabled) {
      delete acc[status.name];
    }

    return acc;
  }, { ...cards });
```
my bad, this this one more time